Final/RFC00XX-Custom-Remote-Connections.md

```diff
@@ -0,0 +1,281 @@
+---
+RFC: RFC00XX
+Author: Paul Higinbotham
+Status: Draft
+Area: Microsoft.PowerShell.Core
+Comments Due: 10/22/2021
+Plan to implement: Yes
+---
+
+# Custom Remote Connections
+
+PowerShell currently supports six types of remote connections.
+
+- WinRM remote connections
+- SSH remote connections
+- Named pipe local connections
+- Local process connections
+- Hyper-V local VM connections (Windows only)
+- Hyper-V local Container connections (Windows only)
+
+But these connections are implemented internally in the PowerShell engine, and adding new connection types requires modifying the PowerShell engine.
+This RFC is a proposal to add support for optional custom remote connections, so that new remoting connection types can be created and used without having to change PowerShell engine code.  
+
+Custom endpoints and listening services are out of scope for this proposal.
+PowerShell remoting does not provide endpoint hosting or connection listeners, and relies on external solutions such as what WinRM and SSH provide.
+
+## Motivation
+
+As a third party contributor or PowerShell team member, I can create a new remoting connection type as a PowerShell module.
+Users can download the module and use it with PowerShell remoting cmdlets.
+
+### Example 1
+
+PowerShell core used to provide an experimental remoting connection from non-Windows machines to Windows machine WinRM endpoints, through an OMI bridge.
+But it is no longer supported and OMI changes break this connection on many non-Windows platforms.  
+
+As a user and community member, I want to create a module that will support and maintain this connection independent of the PowerShell team and without having to submit changes to PowerShell engine code.  
+
+```powershell
+PS /Users/Paul> Install-PSResource -Name PSWSManConnection -Repository PSGallery
+PS /Users/Paul> $session = New-PSWSManPSSession -Computer WindowsMachine1 -Credential Domain\UserName
+
+PowerShell credential request
+Enter your credentials.
+Password for user Domain\UserName: **************
+
+PS /Users/Paul> Enter-PSSession $session
+[WindowsMachine1]: PS C:\> Users\UserName> 
+```
+
+This example installs the `PSWSManConnection` module from the PowerShell Gallery onto a macOS machine.
```

Comment:
Just to clarify: in this example, the PowerShell module publishing and installation is not mandatory, since the module provides New-PSWSManPSSession functions that returns a PSSession object that PowerShell can use. I am personally for an approach that would focus solely on making it possible to construct a PSSession object usable by PowerShell - how that PSSession object is constructed can be complex, but wrapped into a simple function like New-PSWSManSession. In many cases it's possible that the implementation of this function could fit in a single PowerShell file, and it might be possible to just copy/paste it into a new PowerShell terminal without installing anything.

Reply:
Yes, that is the goal of this RFC which is to enable community code to construct a PSSession object that can be used by PowerShell